okhttp/src/test/java/okhttp3/HttpUrlTest.kt

```diff
@@ -560,7 +561,8 @@ open class HttpUrlTest {
         ']'.code
       )
       .override(
-        Encoding.IDENTITY,
+        // java.net.URL got stricter
+        if (PlatformVersion.majorVersion >= 21)  Encoding.SKIP else Encoding.IDENTITY,
```

Comment:
@swankjesse I'm not sure this is right, is there a better way to make assertions here? That it now fails for URL?

Reply:
Lemme attempt a follow-up. I think we need to fix our code.

Reply:
https://github.com/square/okhttp/pull/8363